improver/cli/__init__.py

```diff
@@ -274,15 +283,25 @@ def with_output(wrapped, *args, output=None, compression_level=1, **kwargs):
             printed instead.
         compression_level (int):
             Will set the compression level (1 to 9), or disable compression (0).
-
+        least_significant_digit (int):
+            If specified, the data will be quantized to a number of significant digits, see
+            http://www.esrl.noaa.gov/psd/data/gridded/conventions/cdc_netcdf_standard.shtml
+            for details. When used with `compression level`, this will result in lossy
+            compression.
     Returns:
         Result of calling `wrapped` or None if `output` is given.
     """
     from improver.utilities.save import save_netcdf
 
     result = wrapped(*args, **kwargs)
+    # Clize automatically converts args to match the default value, however, the default value for least_significant_digit is None
+    least_significant_digit = (
+        int(least_significant_digit)
+        if least_significant_digit is not None
+        else least_significant_digit
+    )
```

Comment:
Not needed if option is type annotated (but it should be still tested that the type conversion works properly - current tests won't cover it).
```suggestion
```